sidechain/block-composer/src/block_composer.rs

```diff
@@ -126,12 +129,18 @@ where
 			Error::Other(format!("Failed to encrypt state payload: {:?}", e).into())
 		})?;
 
+		let header =
+			<<SignedSidechainBlock as SignedSidechainBlockTrait>::Block as SidechainBlockTrait>::HeaderType::new(
+				block_number,
+				parent_hash,
+				shard,
+				blake2_256(&payload).into()
```

Comment:
Here also, we need to compute the hash of all block fields, except the header. Ideally, we have only one implementation of computing the block payload hash.